### DISCUSSION TITLE
NULL Filtering not working

### DISCUSSION THREAD

**Discussion by RobertoLanzeni** [spatie/laravel-query-builder#816] [Q&A]
Hi there. 
I'm using this amazing package , but I'm not finding way to filter field as null: 

This is  URL : 
http://127.0.0.1:8001/filterTest?filter[paziente.cognome]=LANZ&filter[stato_pratica]=3&filter[medico.users.name]=Roberto&filter[data_prericovero]=2022

if i set null at latest filter , it returns  anything.
I also used this approach :  AllowedFilter::exact('data_prericovero')->default(null), but anything has changed. 

How can I reach this goal? 
Tanks  Roberto

**Comment by AlexMcowkin** [comment]
it was broken in the last 5th version.
the problem is in QueryBuilderRequest() method

```php

    protected function getRequestData(?string $key = null, $default = null)
    {
        if (config('query-builder.request_data_source') === 'body') {
            return $this->input($key, $default);
        }

        return $this->get($key, $default); // this is new in the 5th version
        
        return $this->query($key, $default); // this was earlier
    }

```
